From f1cbb5539bfffc9391ef1d5d10a3b8974a14451e Mon Sep 17 00:00:00 2001
From: wangzhiqiang <wangzhiqiang95@huawei.com>
Date: Tue, 25 Oct 2022 17:43:06 -0400
Subject: [PATCH 052/595] [Huawei] block:using kabi reserved field for
 EulerOS's log enhance

Offering: HULK
euleros inclusion
category: bugfix
bugzilla: 187868

--------------------------------

EulerOS developed a feature which need to use bio's kabi field, which
could not be changed once released. So here we using this kabi field
to avoid conflict in future.

Signed-off-by: wangzhiqiang <wangzhiqiang95@huawei.com>
---
 include/linux/blk_types.h | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/include/linux/blk_types.h b/include/linux/blk_types.h
index fb72c8de88b1..3e616267425e 100644
--- a/include/linux/blk_types.h
+++ b/include/linux/blk_types.h
@@ -267,7 +267,11 @@ struct bio {
 
 	struct bio_set		*bi_pool;
 
-	KABI_RESERVE(1)
+	/*
+	 * Use the first RESERVE field for EulerOS
+	 * scsi log enhance feature
+	 */
+	KABI_USE(1, u64 bi_start_time_ns)
 	KABI_RESERVE(2)
 	KABI_RESERVE(3)
 	KABI_RESERVE(4)
-- 
2.17.1

